### PR TITLE
[azure][azi-1524] arm template separate api calls for filters

### PR DESCRIPTION
**Pull request by ashwingandhi-ddog**
Saving host filters and app_service_plan filters are each a separate API call to app registration call.
These 2 filter request are dependent on app registration request succeeding and they only happen if the filter values are provided.

Corresponding UIDefinition.json changes.

### CODE REVIEW

File: azure/deploy-to-azure/azure-integration/CreateUIDefinition-managementgroups.json [hunk 1]
```diff
@@ -239,6 +239,17 @@
                 },
                 "visible": true
               },
+              {
+                "name": "appFilters",
```
Comment (gpalmz) [inline azure/deploy-to-azure/azure-integration/CreateUIDefinition-managementgroups.json:243]: Nit: could we name this `appServicePlanFilters`?

Comment (gpalmz) [inline azure/deploy-to-azure/azure-integration/CreateUIDefinition-managementgroups.json]: Nit: could we name this appServicePlanFilters throughout these files?

File: azure/deploy-to-azure/azure-integration/azuredeploy.json [hunk 2]
```diff
@@ -198,7 +201,105 @@
...
+        },
+        {
+            "condition": "[not(empty(parameters('appFilters')))]",
+            "type": "Microsoft.Resources/deploymentScripts",
+            "apiVersion": "2020-10-01",
+            "name": "[concat('datadog-app-filter-script-', parameters('newguid'))]",
+            "location": "[parameters('location')]",
+            "kind": "AzurePowerShell",
+            "dependsOn": [
+                "[resourceId('Microsoft.Resources/deploymentScripts', concat('datadog-integration-script-', parameters('newguid')))]"
```
Comment (gpalmz) [inline azure/deploy-to-azure/azure-integration/azuredeploy.json:264]: If I'm understanding, based on this, both of the filters upserts can be submitted concurrently. We should test to make sure this consistently works and doesn't cause some sort of concurrency-based failure in the backend. If we see issues, we should just make one of the filter upserts depend on the other

**Review by gpalmz — COMMENTED** [review]
One thought about the filter upserts

File: azure/deploy-to-azure/azure-integration/azuredeploy.json (hunk 2 above)
Comment (ashwingandhi-ddog) [reply azure/deploy-to-azure/azure-integration/azuredeploy.json:264]: I had tested this a bunch of times and it works fine keeping the above in mind.

Some thoughts:
- Flask web server can handle multiple requests though it is not very scalable. We would have max 2 requests. It queues them if it is configured as single threaded.
-  the primary key for the `azure_namespace_filters` is `(org_id, tenant_id, client_id, namespace)` so there should not be any issues there

(Also, it's very unlikely that the 2 filter requests would be sent at the same time - given that 2 temporary containers need to be spun up to run these scripts)

Comment (gpalmz) [reply azure/deploy-to-azure/azure-integration/azuredeploy.json:264]: I'm convinced, thanks!

**Review by gpalmz — APPROVED** [review]

**Pull request merged**